ush/python/pygw/src/schema_pkg/build_schema.py

```diff
@@ -0,0 +1,142 @@
+import logging
+from pydoc import locate
+from typing import Dict
+
+import pygw.yaml_file
+from pygw.attrdict import AttrDict
+from pygw.logger import logit
+from pygw.schema import Optional, Schema
+
+# ----
+
+logger = logging.getLogger(__name__.split(".")[-1])
+
+# ----
+
+__all__ = ["BuildSchema"]
+
+# ----
+
+
+class BuildSchema:
+    """
+    Description
+    -----------
+
+    This is the base-class object for setting up and defining the
+    schema for a give application; it is based on
+    https://tinyurl.com/pyutils-schema-build.
+
+    Parameters
+    ----------
+
+    yaml_file: str
+
+        A Python string containg the YAML-formatted file path
+        containing the schema; the YAML-formatted file containing the
+        schema attributes should be formatted similar to the example
+        below.
+
+        required:
+            variable1: bool
+            variable2: float
+            variable3: int
+
+        optional:
+            variable4:
+                type: bool
+                default: False
+            variable5:
+                type: bool
+                default: True
+            variable6:
+                type: float
+                default: 1.0
```

Comment:
Instead of keys `required` and `optional`, I would recommend that a `required` schema attribute be defined.
If the attribute is not present, it is understood that the variable is `optional`.  If `required` is present, its value can be `True|False` making it `Required|Optional`.  If `required` is `False`, then a `default` value must be provided.

E.g.
```
variable1:
    type: bool
     required: True
variable2:
    type: int
    default: 4
    required: False
variable3:
    type: float
    default: 0.25
```

Reply:
Good idea. That makes it much cleaner. I did note that we could have schema options i.e., `Or` or 'And`. I haven't seen a use case for it yet but maybe something to keep in mind.